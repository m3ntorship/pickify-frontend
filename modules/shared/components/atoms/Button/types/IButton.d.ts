import type * as EButton from './EButton';

declare namespace IButton {
  export interface IProps extends IHandlers {
    [property: string]: unknown;
    buttonText?: string | null;
    disabled?: boolean;
    variant: EButton.buttonVariantValues;
    size: EButton.buttonSizeValues;
    leftIcon?: boolean;
    rightIcon?: boolean;
    onlyIcon?: boolean;
    buttonType?: 'button' | 'reset' | 'submit';
  }

  export interface IHandlers {
    onClick?: () => void;
  }
}
export { IButton };
