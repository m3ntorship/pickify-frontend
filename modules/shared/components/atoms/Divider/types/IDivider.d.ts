declare namespace IDivider {
  export interface IProps {
    type: string;
    length: string;
  }
}
export { IDivider };
