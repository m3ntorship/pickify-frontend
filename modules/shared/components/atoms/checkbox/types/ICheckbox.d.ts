declare namespace ICheckbox {
  export interface ICheckBoxProps {
    disabled: boolean;
    size: 'Default' | 'Small';
    onMockClick?: () => void;
  }
  export interface ICheckBoxSvgProps {
    style: string;
  }
}
export { ICheckbox };
