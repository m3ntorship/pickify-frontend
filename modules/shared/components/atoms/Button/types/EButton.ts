export enum buttonVariantValues {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  QUATERNARY = 'quaternary',
  TEXT = 'text',
  LIKE = 'like',
  DISLIKE = 'dislike',
  DELETE = 'delete',
}
export enum buttonSizeValues {
  FULLWIDTH = 'fullWidth',
  XLARGE = 'xlarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  NORMAL = 'normal',
  SMALL = 'small',
}
