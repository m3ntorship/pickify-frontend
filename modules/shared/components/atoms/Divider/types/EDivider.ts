export enum DividerType {
  Horizontal = 'horizontal',
  Vertical = 'vertical',
}
