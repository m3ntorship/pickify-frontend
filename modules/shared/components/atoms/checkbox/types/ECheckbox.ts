export enum Size {
  Default = 'Default',
  Small = 'Small',
}
