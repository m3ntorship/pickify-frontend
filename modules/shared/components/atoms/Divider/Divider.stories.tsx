import React from 'react';
import type { Story, Meta } from '@storybook/react';
import Divider from './Divider';
import type { IDivider } from './types/IDivider';
import * as EDivider from './types/EDivider';

export default {
  title: 'Atoms/Divider',
  component: Divider,
  argTypes: {
    type: {
      control: {
        type: 'radio',
        options: [
          EDivider.DividerType.Horizontal,
          EDivider.DividerType.Vertical,
        ],
      },
    },
  },
} as Meta;

const Template: Story<IDivider.IProps> = (args) => <Divider {...args} />;

export const textInput = Template.bind({});
textInput.args = {
  type: EDivider.DividerType.Horizontal,
  length: '300px',
};
