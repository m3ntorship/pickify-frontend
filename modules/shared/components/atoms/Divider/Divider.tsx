import React from 'react';
import className from 'classnames';
import type { FC, ReactElement, CSSProperties } from 'react';
import type { IDivider } from './types/IDivider';
import * as EDivider from './types/EDivider';

const Divider: FC<IDivider.IProps> = ({ type, length }): ReactElement => {
  const dividerType = className('bg-grey-shd6', {
    'h-px': type === EDivider.DividerType.Horizontal,
    'w-px': type === EDivider.DividerType.Vertical,
  });

  const getDividerLength = (): CSSProperties => {
    return type === EDivider.DividerType.Horizontal
      ? { width: length }
      : { height: length };
  };
  return (
    <>
      <div className={dividerType} style={getDividerLength()} />
    </>
  );
};

export default Divider;
