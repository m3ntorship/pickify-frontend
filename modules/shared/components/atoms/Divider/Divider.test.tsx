import React from 'react';
import renderer from 'react-test-renderer';
import Divider from './Divider';
import * as EDivider from './types/EDivider';

describe('Divide', () => {
  it('should render horizontally when given horizontal type property and appropriate length', () => {
    const tree = renderer
      .create(<Divider type={EDivider.DividerType.Horizontal} length="300px" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render vertically when given vertical type property and appropriate length', () => {
    const tree = renderer
      .create(<Divider type={EDivider.DividerType.Vertical} length="300px" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
