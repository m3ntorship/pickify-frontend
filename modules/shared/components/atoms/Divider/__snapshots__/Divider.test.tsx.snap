// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divide should render horizontally when given horizontal type property and appropriate length 1`] = `
<div
  className="bg-grey-shd6 h-px"
  style={
    Object {
      "width": "300px",
    }
  }
/>
`;

exports[`Divide should render vertically when given vertical type property and appropriate length 1`] = `
<div
  className="bg-grey-shd6 w-px"
  style={
    Object {
      "height": "300px",
    }
  }
/>
`;
